src/spec.c

```diff
@@ -1510,13 +1544,18 @@ t_fieldMask IndexSpec_ParseFieldMask(IndexSpec *sp, RedisModuleString **argv, in
   return ret;
 }
 
+// Assuming the spec is properly locked before calling this function.
+// TODO: multithreaded: verify that the spec is locked
 void IndexSpec_InitializeSynonym(IndexSpec *sp) {
   if (!sp->smap) {
     sp->smap = SynonymMap_New(false);
     sp->flags |= Index_HasSmap;
   }
 }
 
+// Assuming the spec is properly locked before calling this function.
```

Comment:
Seems `IndexSpec_ParseStopWords` is dead code
(actually handled by https://github.com/RediSearch/RediSearch/blob/866eb74244e2071bbac1ca2e83b1dfb9a0c68159/src/spec.c#L1128)